test/http.js

```diff
@@ -56,6 +61,16 @@ test('error with code', async t => {
 	}
 });
 
+test('304 doesn\'t throw', async t => {
+	try {
+		const response = await got(`${s.url}/304`);
+		t.is(response.statusCode, 304);
+		t.is(response.body, '');
+	} catch (err) {
+		t.fail('Exception was thrown');
+	}
```

Comment:
Use `t.notThrows` rather than try/catch:

```js
const p = got(`${s.url}/304`)
await t.notThrows(p);
const response = await p;
t.is(response.statusCode, 304);
t.is(response.body, '');
```